code/game/machinery/igniter.dm

```diff
@@ -43,26 +101,41 @@
 	on = !( on )
 	update_appearance()
 
-/obj/machinery/igniter/process() //ugh why is this even in process()?
-	if (on && !(machine_stat & NOPOWER) )
-		var/turf/location = loc
-		if (isturf(location))
-			location.hotspot_expose(1000,500,1)
-	return 1
+/// Have to process to ignite any gas that comes in the turf
+/obj/machinery/igniter/process()
+	if(!on)
+		return 1
```

Comment:
- Is there a reason why these `return 1`? Seems wrong, especially in process
- This should really stop processing when not on rather than no-op process

Reply:
No reason just wanted to make the code shorter but yeah I'll make it stop processing